File: BodyFacts/CoreDataUtils.m
//
//  CoreDataUtils.m
//  RoseTasks
//
//  Created by Tracy Richard on 7/20/16.
//  Copyright © 2016 Jack Richard. All rights reserved.
//

#import "CoreDataUtils.h"
#import "AppDelegate.swift"


@implementation CoreDataUtils

static NSManagedObjectContext* _managedObjectContext;

+ (NSManagedObjectContext*) managedObjectContext {
    if (_managedObjectContext == nil) {
        AppDelegate* appDelegate = [[UIApplication sharedApplication] delegate];
        
        _managedObjectContext = appDelegate.managedObjectContext;
       
    }
    return _managedObjectContext;
}
+ (void) saveContext {
    NSError *error = nil;
    NSManagedObjectContext* context = [CoreDataUtils managedObjectContext];
    
    if ([context hasChanges] && ![context save:&error]) {
        NSLog(@"Unresolved error %@, %@", error, [error userInfo]);
        abort();
    }
}


@end

File: BodyFacts/CoreDataUtils.swift
//
//  CoreDataUtils.swift
//  BodyFacts
//
//  Created by Tracy Richard on 7/27/16.
//  Copyright © 2016 Jack Richard. All rights reserved.
//
import UIKit
import CoreData
class CoreDataUtils: NSObject {
    var _managedObjectContext : NSManagedObjectContext?
    
    func managedObjectContext () -> NSManagedObjectContext {
        if (_managedObjectContext == nil) {
            let appDelegate = UIApplication.sharedApplication().delegate as! AppDelegate
            _managedObjectContext = appDelegate.managedObjectContext;
        }
        return _managedObjectContext!;
    }
    
    func saveContext () {
        do {
            try self._managedObjectContext?.save()
        } catch {
            print("error")
            abort()
        }
    }
    
    
    
}